src/server.c

```diff
@@ -4390,11 +4390,14 @@ int writeCommandsDeniedByDiskError(void) {
     {
         return DISK_ERROR_TYPE_RDB;
     } else if (server.aof_state != AOF_OFF) {
+        if (server.aof_last_write_status == C_ERR) {
+            return DISK_ERROR_TYPE_AOF;
+        }
+        /* AOF fsync error. */
         int aof_bio_fsync_status;
         atomicGet(server.aof_bio_fsync_status,aof_bio_fsync_status);
-        if (server.aof_last_write_status == C_ERR ||
-            aof_bio_fsync_status == C_ERR)
-        {
+        if (aof_bio_fsync_status == C_ERR) {
+            atomicGet(server.aof_bio_fsync_errno,server.aof_last_write_errno);
```

Comment:
why do we need that variable? we're not using it (no sense in showing it in INFO). the errno is already logged to the logfile.

Reply:
we need to show the error info for users'  requests
https://github.com/redis/redis/blob/e138698e54e97bfaababf56507026bf92dd4deb4/src/scripting.c#L652
https://github.com/redis/redis/blob/1eb85249e7e9dafe4fbd023771e53c9a804c0a2f/src/server.c#L4122

Reply:
ohh, i missed the fact you read from one server variable, and write into the other.